bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/ThingHandlerPartition.java

```diff
@@ -0,0 +1,116 @@
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.caddx.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.caddx.internal.CaddxBindingConstants;
+import org.openhab.binding.caddx.internal.CaddxEvent;
+import org.openhab.binding.caddx.internal.CaddxMessage;
+import org.openhab.binding.caddx.internal.CaddxMessageType;
+import org.openhab.binding.caddx.internal.CaddxProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a class for handling a Partition type Thing.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@NonNullByDefault
+public class ThingHandlerPartition extends CaddxBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ThingHandlerPartition.class);
+
+    /**
+     * Constructor.
+     *
+     * @param thing
+     */
+    public ThingHandlerPartition(Thing thing) {
+        super(thing, CaddxThingType.PARTITION);
+    }
+
+    @Override
+    public void updateChannel(ChannelUID channelUID, String data) {
+        // All Zone channels are OnOffType
+        OnOffType onOffType;
+
+        onOffType = ("true".equals(data)) ? OnOffType.ON : OnOffType.OFF;
+        updateState(channelUID, onOffType);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand(): Command Received - {} {}.", channelUID, command);
+
+        String cmd = null;
+        String data = null;
+        CaddxBridgeHandler bridgeHandler = getCaddxBridgeHandler();
+        if (bridgeHandler == null) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            if (channelUID.getId().equals(CaddxBindingConstants.PARTITION_ARMED)) {
+                cmd = CaddxBindingConstants.PARTITION_STATUS_REQUEST;
+                data = String.format("%d", getPartitionNumber() - 1);
+            } else {
+                return;
+            }
+        } else if (channelUID.getId().equals(CaddxBindingConstants.PARTITION_PRIMARY_COMMAND)
+                || channelUID.getId().equals(CaddxBindingConstants.PARTITION_SECONDARY_COMMAND)) {
+            cmd = channelUID.getId();
+            data = command.toString() + "," + String.format("%d", getPartitionNumber() - 1);
+        } else {
+            throw new IllegalArgumentException("Unknown command");
```

Comment:
Can you make sure you don't throw exceptions when you encounter an unknown command. Logging should be used instead.

Reply:
Done